relevanceai/vector_tools/cluster.py

```diff
@@ -332,6 +332,112 @@ def fit_transform(
         return cluster_labels
 
 
+class HierarchicalClusterer(DensityCluster):
+    def __init__(
+        self,
+        n_clusters: Union[int, None] = None,
+        affinity: str = 'euclidean',
+        memory = Memory(cachedir=None),
+        compute_full_tree: Union[str, bool] ='auto',
+        linkage: str = 'ward',
+        distance_threshold: Union[float, None] = None,
+        compute_distances: bool = False,
+        dendrogram_plot_args: Union[dict, None] = None,
+    ):
+        self.n_clusters = n_clusters
+        self.affinity = affinity
+        self.memory = memory
+        self.compute_full_tree = compute_full_tree
+        self.linkage = linkage
+        self.distance_threshold = distance_threshold
+        self.compute_distances = compute_distances
+        self.dendrogram_plot_args = dendrogram_plot_args
+
+    def fit_transform(
+        self,
+        vectors: np.ndarray
+    ) -> np.ndarray:
+        try:
+            from sklearn.cluster import AgglomerativeClustering
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                f"{e}\nInstall Hierarchical Clustering\n \
+                pip install -U relevanceai[hierarchical]"
+            )
+
+        agg = AgglomerativeClustering(
```

Comment:
keep model as an attribute once fitted in case we need something else and don't want to re-fit